statsmodels/tsa/_innovations.pyx

```diff
@@ -0,0 +1,166 @@
+    >>> nobs = activity.shape[0]
+    >>> theta, sigma2  = innovations_algo(acov[:4], nobs=nobs)
+
+    See Also
+    --------
+    innovations_filter
+
+    References
+    ----------
+    .. [*] Brockwell, P.J. and Davis, R.A., 2016. Introduction to time series
+        and forecasting. Springer.
+    """
+    cdef double[::1] v, _acov
+    cdef double[:, ::1] theta
+    cdef Py_ssize_t i, j, k, max_lag
+    cdef double sub, _rtol
+
+    acov = np.ascontiguousarray(np.squeeze(np.asarray(acov)), dtype=np.double)
+    if acov.ndim != 1:
+        raise ValueError('acov must be 1-d or squeezable to 1-d.')
+    _acov = acov
+    rtol = 0.0 if rtol is None else rtol
+    if not isinstance(rtol, float):
+        raise ValueError('rtol must be a non-negative float or None.')
+    _rtol = rtol
+    if nobs is not None and (nobs != int(nobs) or nobs < 1):
+        raise ValueError('nobs must be a positive integer')
+    n = acov.shape[0] if nobs is None else int(nobs)
+    max_lag = int(np.max(np.argwhere(acov != 0)))
+
+    v = np.zeros(n + 1)
+    v[0] = _acov[0]
+    # Retain only the relevant columns of theta
+    theta = np.zeros((n + 1, max_lag + 1))
+    for i in range(1, n):
+        for k in range(max(i - max_lag, 0), i):
+            sub = 0
+            for j in range(max(i - max_lag, 0), k):
+                sub += theta[k, k - j] * theta[i, i - j] * v[j]
+            theta[i, i - k] = 1. / v[k] * (_acov[i - k] - sub)
+        v[i] = _acov[0]
+        for j in range(max(i - max_lag, 0), i):
+            v[i] -= theta[i, i - j] ** 2 * v[j]
+        # Break if v has converged
+        if i >= 10:
+            if v[i - 10] - v[i] < v[0] * _rtol:
+                # Forward fill all remaining values
+                v[i + 1:] = v[i]
+                theta[i + 1:] = theta[i]
+                break
+    return np.asarray(theta[:n, 1:]), np.asarray(v[:n])
+
+
+def innovations_filter(endog, theta):
+    """
+    innovations_filter(endog, theta)
+
+    Filter observations using the innovations algorithm
+
+    Parameters
+    ----------
+    endog : array_like
+        The time series to filter (nobs,). Should be demeaned if not mean 0.
+    theta : ndarray
+        Innovation coefficients of MA representation. Array must be (nobs, q)
+        where q order of the MA.
+
+    Returns
+    -------
+    resid : ndarray
+        Array of filtered innovations
+
+    Examples
+    --------
+    >>> import statsmodels.api as sm
+    >>> data = sm.datasets.macrodata.load_pandas()
+    >>> rgdpg = data.data['realgdp'].pct_change().dropna()
+    >>> acov = sm.tsa.acovf(rgdpg)
+    >>> nobs = activity.shape[0]
+    >>> theta, sigma2  = innovations_algo(acov[:4], nobs=nobs)
+    >>> resid = innovations_filter(rgdpg, theta)
+
+    See Also
+    --------
+    innovations_algo
+
+    References
+    ----------
+    .. [*] Brockwell, P.J. and Davis, R.A., 2016. Introduction to time series
+        and forecasting. Springer.
+    """
```

Comment:
Does this make it into the docs if it’s in cython?

Reply:
The doc build is clean which indicates that these show up in the docs. There are other Cythonized functions that show up in docs.  ARAIK the only important change is that the cythonized version needs to provide the function sig in the first line of the docstring.